lib/datadog/tracing/tracer.rb

```diff
@@ -494,7 +494,11 @@ def sample_span(trace_op, span)
       def flush_trace(trace_op)
         begin
           trace = @trace_flush.consume!(trace_op)
-          write(trace) if trace && !trace.empty?
+          if trace && !trace.empty?
+            # check if trace is not sampled
+            sample_trace(trace) unless trace.sampled?
```

Comment:
```suggestion
            sample_trace(trace)
```
sample_trace method checks if the sampling priority is already set